cpp/independent_set/ExactRecursiveIndependentSet.cpp

```diff
@@ -0,0 +1,755 @@
+/*
+ * ExactRecrusvieIndependentSet.cpp
+ *
+ *  Created on: 26.08.2023
+ *      Author: Artur Salawa
+ */
+
+#include <independent_set/IndependentSet.hpp>
+
+#include <networkit/auxiliary/BucketPQ.hpp>
+#include <networkit/graph/GraphTools.hpp>
+
+#include <limits>
+
+namespace Koala {
+
+void RecursiveIndependentSet::run() {
+    std::vector<NetworKit::node> result = recursive();
+    graph->forNodes([&](NetworKit::node v) {
+        independentSet[v] = false;
+    });
+    for (auto node : result) {
+        independentSet[node] = true;
+    }   
+    hasRun = true;
+}
+
+std::vector<NetworKit::node> Mis1IndependentSet::recursive() {
+    if (graph->isEmpty()) {
+        return {};
+    }
+    NetworKit::node v = getMinimumDegreeNode();
+    std::vector<NetworKit::node> selectOneOf = getNeighborsPlus(v);
```

Comment:
In cases like this, I encourage to use "auto" - less bugs, less refactoring in the future for next developers.